src/ray/protobuf/gcs.proto

```diff
@@ -485,6 +485,49 @@ message PlacementGroupLoad {
   repeated PlacementGroupTableData placement_group_data = 1;
 }
 
+message PlacementGroupStats {
+  // The scheduling state of the placement group.
+  // This is purely for observability and the
+  // information could not be consistent
+  // if GCS restarts.
+  enum SchedulingState {
+    // The initial state.
+    WAITING_FOR_SCHEDULING = 0;
+    // Placement group creation failed because there was
+    // no nodes that satisfy resources.
+    NO_RESOURCES = 1;
+    // Placement group creation failed because there was
+    // an issue commiting resources to nodes.
+    FAILED_TO_COMMIT_RESOURCES = 2;
+    // Scheduling has cancelled because it is removed.
+    REMOVED = 3;
+    // Scheduling has finished properly.
+    FINISHED = 4;
+    // Placement group creation has failed because the
+    // requested resources are infeasible.
+    INFEASIBLE = 5;
+  }
+  // The time creation request is first received.
+  int64 creation_request_received_ns = 2;
+  // The time scheduling is started after queuing.
+  int64 scheduling_started_time_ns = 3;
+  // The time from scheduling started <-> scheduling succeeds.
+  double scheduling_latency_us = 4;
```

Comment:
Ah, I used microsecond at first, so I used double to catch microseconds level values. I think since I am using microseconds, int64_t is probably sufficient (creation usually takes about 1 ms it seems like, so catching diff in nano seconds probably has no value.)